OMEdit/OMEditLIB/Element/Element.cpp

```diff
@@ -3443,16 +3447,26 @@ void Element::resizeElement(QPointF newPosition)
   QTransform tmpTransform = QTransform().translate(pivot.x(), pivot.y()).rotate(0)
       .scale(mXFactor, mYFactor)
       .translate(-pivot.x(), -pivot.y());
+  // save previous scaling
+  qreal sxPre, syPre;
+  getScale(&sxPre, &syPre);
   setTransform(mTransform * tmpTransform);
   // set the final resize on component.
   QPointF extent1, extent2;
   qreal sx, sy;
   getScale(&sx, &sy);
-  extent1.setX(sx * boundingRect().left());
-  extent1.setY(sy * boundingRect().top());
-  extent2.setX(sx * boundingRect().right());
-  extent2.setY(sy * boundingRect().bottom());
-  mTransformation.setOrigin(scenePos());
+  qreal kx = sx/sxPre;
+  qreal ky = sy/syPre;
+  qreal x = pivot.x() + (mTransformation.getOrigin().x() - pivot.x()) * kx;
+  qreal y = pivot.y() + (mTransformation.getOrigin().y() - pivot.y()) * ky;
+  ExtentAnnotation currentExtent = mTransformation.getExtent();
+  QPointF currentExtent1 = currentExtent.at(0);
+  QPointF currentExtent2 = currentExtent.at(1);
+  extent1.setX(currentExtent1.x() * kx);
+  extent1.setY(currentExtent1.y() * ky);
+  extent2.setX(currentExtent2.x() * kx);
+  extent2.setY(currentExtent2.y() * ky);
+  mTransformation.setOrigin(QPointF(x, y));
```

Comment:
Something is off with this code. Try resizing the components. Resize them in a way that they flip horizontally and vertically.